src/marqo/s2_inference/reranking/rerank.py

```diff
@@ -18,24 +23,60 @@ def rerank_search_results(search_result: Dict, query: str, model_name: str, devi
         overwrite_original_scores_highlights (bool, optional): _description_. Defaults to True.
     """
 
-    # TODO add in routing based on model type
+    # check the search_results have the searchable attribute before proceeding
+    # skip reranking if the results do not contain the field
+    if not _check_searchable_fields_in_results(search_results=search_result, searchable_fields=searchable_attributes):
+        return search_result
+
     if 'owl' in model_name.lower():
-        reranker = ReRankerOwl(model_name=model_name, device=device, image_size=(240,240))
-        reranker.rerank(query=query, results=search_result, image_attributes=searchable_attributes)
+        # owl needs the image location, while the text based ones can handle different number of fields but concat the text
+        if searchable_attributes in (None, [], (), ''):
+            raise RerankerError(f"found searchable_attributes={searchable_attributes} but expected list of strings for {model_name}")
+        
+        if len(searchable_attributes) > 1:
+            logger.info(f"currently only a single attribute can be reranked over for {model_name}. taking the first field {[searchable_attributes[0]]} from {searchable_attributes}")
+            searchable_attributes = [searchable_attributes[0]]
+
```

Comment:
This needs to be communicated to users. For example:
- a warning accompanying a multiple fields rerank request
- an error if they attempt over multiple fields
- The API specs prevent it